src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -579,7 +579,8 @@ public void RemoveAt(int index)
             /// <param name="length">The number of elements to remove.</param>
             public void RemoveRange(int index, int length)
             {
-                Requires.Range(index >= 0 && index + length <= _count, nameof(index));
+                Requires.Range(index >= 0 && index <= _count, nameof(index));
+                Requires.Range(length >= 0 && index <= _count - length, nameof(length));
```

Comment:
nit, so that the arg being validated is on the lhs
```suggestion
                Requires.Range(length >= 0 && length <= _count - index, nameof(length));
```